packages/devtools_app/lib/src/app.dart

```diff
@@ -162,6 +162,8 @@ class DevToolsAppState extends State<DevToolsApp> {
 
     // Always return the landing screen if there's no VM service URI.
     if (vmServiceUri?.isEmpty ?? true) {
+      // Ensure we disconnect from any previously connected apps.
+      serviceManager.manuallyDisconnect();
```

Comment:
building a widget shouldn't have side effects like this. Is there a different fix?

Reply:
Moved disconnect calls into router. Fixing tests, then will land so I can get this into the dev release.